okhttp-testing-support/src/main/java/okhttp3/PlatformRule.kt

```diff
@@ -55,55 +59,107 @@ open class PlatformRule @JvmOverloads constructor(
 
   fun isJdk9() = getPlatformSystemProperty() == JDK9_PROPERTY
 
-  fun isJdk8() = getPlatformSystemProperty() == DEFAULT_PROPERTY
+  fun isJdk8() = getPlatformSystemProperty() == JDK8_PROPERTY
+
+  fun isJdk8Alpn() = getPlatformSystemProperty() == JDK8_ALPN_PROPERTY
+
+  fun hasHttp2Support() = !isJdk8()
 
   fun assumeConscrypt() {
-    assumeThat(getPlatformSystemProperty(), equalTo(CONSCRYPT_PROPERTY))
+    assumeThat(getPlatformSystemProperty(), equalTo(
+        CONSCRYPT_PROPERTY))
   }
 
   fun assumeJdk9() {
-    assumeThat(getPlatformSystemProperty(), equalTo(JDK9_PROPERTY))
+    assumeThat(getPlatformSystemProperty(), equalTo(
+        JDK9_PROPERTY))
   }
 
   fun assumeJdk8() {
-    assumeThat(getPlatformSystemProperty(), equalTo(DEFAULT_PROPERTY))
+    assumeThat(getPlatformSystemProperty(), equalTo(
+        JDK8_PROPERTY))
+  }
+
+  fun assumeJdk8Alpn() {
+    assumeThat(getPlatformSystemProperty(), equalTo(
+        JDK8_ALPN_PROPERTY))
+  }
+
+  fun assumeHttp2Support() {
+    assumeThat(getPlatformSystemProperty(), not(
+        JDK8_PROPERTY))
   }
 
   fun assumeNotConscrypt() {
-    assumeThat(getPlatformSystemProperty(), not(CONSCRYPT_PROPERTY))
+    assumeThat(getPlatformSystemProperty(), not(
+        CONSCRYPT_PROPERTY))
   }
 
   fun assumeNotJdk9() {
-    assumeThat(getPlatformSystemProperty(), not(JDK9_PROPERTY))
+    assumeThat(getPlatformSystemProperty(), not(
+        JDK9_PROPERTY))
   }
 
   fun assumeNotJdk8() {
-    assumeThat(getPlatformSystemProperty(), not(DEFAULT_PROPERTY))
+    assumeThat(getPlatformSystemProperty(), not(
+        JDK8_PROPERTY))
+  }
+
+  fun assumeNotJdk8Alpn() {
+    assumeThat(getPlatformSystemProperty(), not(
+        JDK8_ALPN_PROPERTY))
+  }
+
+  fun assumeNotHttp2Support() {
+    assumeThat(getPlatformSystemProperty(), equalTo(
+        JDK8_PROPERTY))
   }
 
   fun assumeJettyBootEnabled() {
     assumeTrue("ALPN Boot not enabled", isAlpnBootEnabled())
   }
 
   companion object {
-    val PROPERTY_NAME = "okhttp.platform"
-    val CONSCRYPT_PROPERTY = "conscrypt"
-    val DEFAULT_PROPERTY = "default"
-    val JDK9_PROPERTY = "jdk9"
+    const val PROPERTY_NAME = "okhttp.platform"
+    const val CONSCRYPT_PROPERTY = "conscrypt"
+    const val JDK9_PROPERTY = "jdk9"
+    const val JDK8_ALPN_PROPERTY = "jdk8alpn"
+    const val JDK8_PROPERTY = "jdk8"
 
     init {
       if (getPlatformSystemProperty() == CONSCRYPT_PROPERTY && Security.getProviders()[0].name != "Conscrypt") {
+        if (!Conscrypt.isAvailable()) {
+          System.err.println("Warning: Conscrypt not available")
```

Comment:
Does this need to throw? Or we expect it to just fail somewhere else down the line maybe.

Reply:
I think you are right.

But to start with I didn't really want to change much behaviour, just make things more predictable.  I don't want it throwing new exceptions that we hadn't see before.  I think System.err will be a good pointer in an IDE console that something isn't right.

Let's land as is and see how the experience is in CI and IDEs, at least now it's behind this API and the tests are quite intentional, so you can search for each assumption method.